### PR TITLE
[~] - update ami to amazon linux 2

### PR DESCRIPTION
**Pull request by oreststetsiak**
update ami to amazon linux 2

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -152,7 +158,7 @@ resource "aws_security_group" "rabbitmq_nodes" {
 
 resource "aws_launch_configuration" "rabbitmq" {
   name                 = local.cluster_name
-  image_id             = data.aws_ami_ids.ami.ids[0]
+  image_id             = data.aws_ami.amazon-linux-2.id
```
Comment (rraub) [inline main.tf:161]:
```suggestion
  image_id             = data.aws_ami_ids.amazon-linux-2.ids[0]
```

File: main.tf [hunk 2]
```diff
@@ -5,12 +5,18 @@ data "aws_vpc" "vpc" {
 data "aws_region" "current" {
 }
 
-data "aws_ami_ids" "ami" {
-  owners = ["amazon"]
+data "aws_ami" "amazon-linux-2" {
+  owners      = ["amazon"]
+  most_recent = true
```
Comment (rraub) [inline main.tf:10]:
```suggestion
```
This isn't an valid attribute, https://www.terraform.io/docs/providers/aws/d/ami_ids.html#sort_ascending is probably what you were thinking of but that defaults to descending which is what we want.

**Review by rraub — CHANGES_REQUESTED** [review]

**Comment by rraub** [comment]
Also this ami broke the `find_hosts.sh` script since the `$HOSTNAME` changed from just `ip-10-2-1-82` to `ip-10-2-1-82.ec2.internal` in the newer Amazon Machine Linux 2.  see: https://github.com/smartrent/terraform-aws-rabbitmq/pull/1 if your interested

**Comment by krotkiewicz** [comment]
Hey @oreststetsiak, thank you for PR and @rraub for comments. 

Currently we don't have resources to prepare CI pipeline with some basic tests or to do tests manually. If you have some spare time and could help here, let me know.

**Pull request closed without merging**